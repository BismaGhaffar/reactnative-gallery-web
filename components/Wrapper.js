import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Footer from './Footer'

const Wrapper = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 1em 3em;
  min-height: 100%;
  font-size: 20px;
  margin: 0;
  width: auto;
  height: auto;
  line-height: 1.42857;
  word-wrap: normal;
  color: #333;
  z-index: 1;
  @media (max-width: 481px) {
    font-size: 16px;
    padding: 0;
    line-height: 1.21;
  }
`

function Wrap({ children, footer }) {
  return (
    <Wrapper>
      {children}
      {footer && <Footer />}
    </Wrapper>
  )
}

Wrap.defaultProps = {
  footer: false,
  children: <p style={{ color: '#f55' }}>Nothing to render</p>
}

Wrap.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element
  ]),
  footer: PropTypes.bool
}

export default Wrap
