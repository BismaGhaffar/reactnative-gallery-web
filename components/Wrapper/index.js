import styled from 'styled-components'

const Wrapper = styled.section`
  padding: 1em 7em;
  background: #fff;
  font-family: 'Montserrat', sans-serif;
  min-height: 100%;
  font-size: 19px;
  margin: 0;
  width: auto;
  height: auto;
  line-height: 1.42857;
  word-wrap: normal;
  color: #333;
  z-index: 1;
  background-color: rgb(25, 8, 1);
  @media (max-width: 700px) {
    padding: 1em 3em;
  }
`

export default Wrapper
