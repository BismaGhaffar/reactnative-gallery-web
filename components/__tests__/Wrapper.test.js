import React from 'react'
import renderer from 'react-test-renderer'
import Wrapper from '../Wrapper'

const compToTest = (
  <Wrapper>
    <p>test</p>
  </Wrapper>
)

const compToTest2 = <Wrapper />

const compToTest3 = <Wrapper footer />

it('Wrapper can be created', () => {
  const comp = renderer.create(compToTest)
  expect(comp).toBeDefined()
})

it('<Wrapper /> toMatchSnapshot', () => {
  const tree = renderer.create(compToTest).toJSON()
  expect(tree).toMatchSnapshot()
})

it('<Wrapper /> toMatchSnapshot', () => {
  const tree = renderer.create(compToTest2).toJSON()
  expect(tree).toMatchSnapshot()
})

it('<Wrapper /> toMatchSnapshot', () => {
  const tree = renderer.create(compToTest3).toJSON()
  expect(tree).toMatchSnapshot()
})
