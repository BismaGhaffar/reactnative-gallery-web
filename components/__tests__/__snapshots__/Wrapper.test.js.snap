// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> toMatchSnapshot 1`] = `
<section
  className="Wrapper-s1d4go23-0 goHcGs"
>
  <p>
    test
  </p>
</section>
`;

exports[`<Wrapper /> toMatchSnapshot 2`] = `
<section
  className="Wrapper-s1d4go23-0 goHcGs"
>
  <p
    style={
      Object {
        "color": "#f55",
      }
    }
  >
    Nothing to render
  </p>
</section>
`;

exports[`<Wrapper /> toMatchSnapshot 3`] = `
<section
  className="Wrapper-s1d4go23-0 goHcGs"
>
  <p
    style={
      Object {
        "color": "#f55",
      }
    }
  >
    Nothing to render
  </p>
  <section
    className="Footer-s19zc472-0 koAdlk"
  >
    <a
      className="Footer__LinkStyl-s19zc472-1 jdNJsj"
      href="https://github.com/ReactNativeGallery/reactnative-gallery-web"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
        />
      </svg>
    </a>
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "#fff",
          "opacity": 0.5,
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
    <a
      className="Footer__LinkStyl-s19zc472-1 jdNJsj"
      href="https://twitter.com/rn_gallery"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
        />
      </svg>
    </a>
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "#fff",
          "opacity": 0.5,
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
    <a
      className="Footer__LinkStyl-s19zc472-1 jdNJsj"
      href="https://slack.reactnative.gallery"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.08 9C19.81 1.41 16.54-.35 9 1.92S-.35 7.46 1.92 15 7.46 24.35 15 22.08 24.35 16.54 22.08 9z"
        />
        <line
          x1="12.57"
          x2="16.15"
          y1="5.99"
          y2="16.39"
        />
        <line
          x1="7.85"
          x2="11.43"
          y1="7.61"
          y2="18.01"
        />
        <line
          x1="16.39"
          x2="5.99"
          y1="7.85"
          y2="11.43"
        />
        <line
          x1="18.01"
          x2="7.61"
          y1="12.57"
          y2="16.15"
        />
      </svg>
      <div
        className="Footer__Stats-s19zc472-2 iwkEDr"
      >
        0
        /
        0
      </div>
    </a>
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "#fff",
          "opacity": 0.5,
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
    <a
      className="Footer__LinkStyl-s19zc472-1 jdNJsj"
      href="https://spectrum.chat/reactnative-gallery"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Join the community on Spectrum"
        src="/static/images/spectrum.svg"
        style={
          Object {
            "maxWidth": 18,
          }
        }
      />
    </a>
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": "#fff",
          "opacity": 0.5,
        }
      }
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
    </svg>
    <a
      className="Footer__LinkStyl-s19zc472-1 jdNJsj"
      href="mailto:xcapetir+rng@gmail.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
        />
        <polyline
          points="22,6 12,13 2,6"
        />
      </svg>
    </a>
  </section>
</section>
`;
